### PR TITLE
S3t1 new email form

### PR DESCRIPTION
**Pull request by cr-walton**
New email form, form validation and error messages

### CODE REVIEW

File: src/Components/NewEmailForm/index.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    const validateEmail = (e) => {
+        let emailAdresses = e.target.value.split(' ')
+            emailAdresses.forEach(emailAddress => {
+                if(!emailAddress.match(validEmail) || emailAddress == '' ) {
+                    setEmailAddressValid(false) 
+                   
+                } else {
+                    setEmailAddressValid(true)
+                }
+                console.log(emailAddressValid)
```
Comment (Jdixon04) [inline src/Components/NewEmailForm/index.js]: get rid of this now if it works and you dont need it anymore

File: src/Components/NewEmailForm/index.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    const [emailSubjectValid, setEmailSubjectValid] = useState(null)
+    
+    const validEmail =  /^[a-zA-Z0-9.!#$%&'*+/=?^_`-]+@[a-zA-Z0-9-]+(?:\.[a-zA-Z0-9-]+)*$/
+
+    const validateEmail = (e) => {
+        let emailAdresses = e.target.value.split(' ')
+            emailAdresses.forEach(emailAddress => {
+                if(!emailAddress.match(validEmail) || emailAddress == '' ) {
+                    setEmailAddressValid(false) 
+
```
Comment (Jdixon04) [inline src/Components/NewEmailForm/index.js]: blank space

File: src/Components/NewEmailForm/index.js [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    const validateSubject = (e) => {
+        if(e.target.value === ''){
+            setEmailSubjectValid(false)
+        } else {
+            setEmailSubjectValid(true)
+        }
+    }
+
+    return (
+
```
Comment (Jdixon04) [inline src/Components/NewEmailForm/index.js:54]: blank space

File: src/Components/NewEmailForm/index.js [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+            <div className="mb-4">
+                <input onBlur={validateSubject} className="col form-control" name="subject" type="text" id="subject" placeholder="Subject"></input>
+                <span className='text-danger'>{emailSubjectValid === false ? 'Email subject empty, send anyway?' : ''}</span>
+            </div>
+            <div className="input-group mb-4">
+                <textarea className="form-control" aria-label="With textarea" placeholder="Write your email here"></textarea>
+            </div>
+        </form>
+    )
+
```
Comment (Jdixon04) [inline src/Components/NewEmailForm/index.js]: blank spaces

**Review by Jdixon04 — COMMENTED** [review]
With regards to the form email body section it has both a drag functionality to increase the space to write in and also creates a scroll bar when the text exceeds the text box. Potentially just having a set size for the body box without the drag to increase feature would or the scroll bar option would be better, not both. 

when having multiple emails or a large subject exceeds the size of the input boxes it becomes quite difficult to navigate back to the beginning of the box on mobile due to the input boxes not being responsive to the size of the text input, maybe look into this issue.  

This input should work but i still get an error? a single email clears the error but this throws it up, needs investigation.
Screenshot 2022-08-31 at 12 39 45

**Review by aidanmaskell — APPROVED** [review]

**Pull request merged**